HelpSource/Reference/AudioDeviceSelection.schelp

```diff
@@ -0,0 +1,181 @@
+::
+LIST::
+##In the teletype::Audio Devices:: window click on the Plus button on the bottom left and choose teletype::Create Aggregate Device::.
+##You can change its name to something specific if desired, for example "InternalAndMOTU".
+##Check the Use column on the right for the devices you want to combine.
+::
+Now you need to tell SuperCollider to use your new aggregate device.
+code::
+Server.default.options.device = "Aggregate Device"; //or the name you have specified in the previous step
+::
+
+After (re)booting the server (code::Server.default.reboot::) you should see in the post window that it now uses the Aggregate Device instead of system defaults:
+teletype::
+[...]
+"Aggregate Device" Input Device
+[...]
+"Aggregate Device" Output Device
+::
+
+Note that when you specify a sound device through SuperCollider's code::ServerOptions::, there is no need to use the aggregate device as the system's default device.
+
+
+section:: Linux
+
+By default SuperCollider on Linux uses JACK, and the audio device selection is managed by the JACK server. teletype::ServerOptions:: cannot override JACK's selection of audio hardware.
+
+subsection:: Setup with JACK server
+note::The SuperCollider server is considered a JACK emphasis::client::. In the following paragraph the term emphasis::client:: will refer to SuperCollider server, from the perspective of JACK. ::
+
+When the server is compiled for JACK natively, the code::ServerOption::'s teletype::device:: can be used in two ways:
+list::
+## to set the client name to register with JACK:
+code::
+Server.default.options.device = "my_synth";
+::
+## to use a specific JACK server, as well as set the client name:
+code::
+Server.default.options.device = "JACKServerName:scsynthName";
+::
+::
+A code::nil:: device is equivalent to code::Server.default.options.device = "default:SuperCollider";::
+
+The JACK connections can be configured via the environment variables teletype::SC_JACK_DEFAULT_INPUTS:: and
+teletype::SC_JACK_DEFAULT_OUTPUTS::. The format is either a string that specifies another jack client or a comma-separated list of jack ports.
+
+code::
+// connect first to input channels with system
+"SC_JACK_DEFAULT_INPUTS".setenv("system:capture_1,system:capture_2");
+
+// connect all output channels with system
+"SC_JACK_DEFAULT_OUTPUTS".setenv("system");
+::
+
+section:: Windows
+
+On Windows there are multiple audio driver APIs (e.g. teletype::MME::, teletype::WDM-KS::, teletype::ASIO:: etc.) that can be used to communicate with audio devices. The API (listed before the device name) needs to match between the input and the output, for example:
+code::
+o = Server.default.options;
+o.inDevice_("Windows WDM-KS : Microphone");
+o.outDevice_("Windows WDM-KS : Speakers");
+Server.default.reboot;
+::
+On Windows you strong::cannot:: programmatically obtain the list of available audio devices. However, all the devices are listed in the post window when the server is booting: code::Server.default.boot::
```

Comment:
is this impossible in the user sense or the developer sense (asking for my own benefit)? how much effort would that take?

Reply:
User sense. I have no idea how much effort.

Currently `int prListAudioDevices(struct VMGlobals *g, int numArgsPushed)` is defined in SC_CoreAudioPrim.cpp. Windows would need a PortAudio equivalent.

Reply:
looks fairly easy; I wouldn't mind doing that after the codebase reformat has landed

Reply:
Is the inconsistency purely in the way the names are presented, or is it more substantial than that? Thank you for the detailed explanation btw

Reply:
By inconsistency I mean the (possible) one between scsynth, supernova and sclang. Right now the scenarios are:

1. macOS - the desired backend is coreaudio (but it is possible to build with portaudio too). However, even if coreaudio is selected, supernova still uses portaudio. Coreaudio and PortAudio use slightly different names for devices, so currently AFAIR ```ServerOptions.devices``` does not provide proper names for use with supernova
2. windows - I think PortAudio is the only option, so as long as the 3 components (lang and synths) use that we should be OK (note that my PR makes sure we list APIs before the device name, so ServerOption.devices would need to mimic that)
3. linux - jack is the only supported configuration so this does not apply, unless one builds with PortAudio (if possible) and then it _should_ be consistent in the same way as it would be on Windows

Because of the lack of native coreaudio support in supernova and the general usability issues with PortAudio, I was wondering if there is another framework that could replace PortAudio and came across [rtAudio](https://github.com/thestk/rtaudio) - but I'm not sure it would be a better solution (looks better in specs though). One nice (?) possible outcome is that we wouldn't absolutely need to use JACK on linux, but I'm not sure this is as much of interest.